back/src/invoice/entity/Invoice.ts

```diff
@@ -0,0 +1,35 @@
+import { Column, Entity, JoinColumn, ManyToOne, OneToMany, OneToOne, PrimaryGeneratedColumn } from "typeorm";
+import { Field, ObjectType } from "type-graphql";
+import { Reservation } from "../../reservation/entity/Reservation";
+import { User } from "../../user/entity/User";
+import { UserBilling } from "../UserBilling";
+
+@ObjectType()
+@Entity()
+export class Invoice {
+    @Field()
+    @PrimaryGeneratedColumn("uuid")
+    id: string;
+
+    @Field(() => Reservation)
+    @OneToOne(() => Reservation, (reservation) => reservation.invoice)
+    @JoinColumn()
+    reservation: Reservation;
+
+    @Field(() => User)
+    @ManyToOne(() => User, (user) => user.invoices)
+    user: User;
+
+    @Field(() => UserBilling)
+    @OneToOne(() => UserBilling, (userBilling) => userBilling.invoice, {
```

Comment:
C'est vraiment une OneToOne ? une même adresse de facturation peut apparaitre sur plusieurs factures différentes (d'un même client) --> One to Many